pkg/3scale/amp/component/system.go

```diff
@@ -869,11 +928,24 @@ func (system *System) systemConfigVolumeMount() v1.VolumeMount {
 	}
 }
 
+func (system *System) s3CredsProjectedVolumeMount() v1.VolumeMount {
+	return v1.VolumeMount{
+		Name:      S3StsCredentialsSecretName,
+		ReadOnly:  true,
+		MountPath: S3StsCredentialsProjectedVolumeMountPath,
```

Comment:
We are mounting the file with the token in a fixed path. And then we are passing the path to the application a value that is read from a secret
```go
result = append(result,
				helper.EnvVarFromSecret(AwsRoleArn, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsRoleArn),
				helper.EnvVarFromSecret(AwsWebIdentityTokenFile, system.Options.S3FileStorageOptions.ConfigurationSecretName, AwsWebIdentityTokenFile),
			)
```

Isn't this prone to errors?

Reply:
@eguzki pushed a draft change to improve this if you have time to take a look.

Reply:
small change done - mount path = tokenPath  - tokenName ("token")